.github/workflows/ld.yml

```diff
@@ -144,27 +150,37 @@ jobs:
     steps:
       -
         name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       -
         name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
+        uses: docker/setup-qemu-action@v3
       -
         name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-        with:
-          driver-opts: image=${{ env.BUILDKIT_IMAGE }}
+        uses: docker/setup-buildx-action@v3
       -
         name: Build
-        uses: docker/bake-action@v2
+        uses: docker/bake-action@v4
         with:
           targets: ${{ matrix.target }}
           set: |
             *.cache-from=type=gha,scope=${{ matrix.target }}
             *.cache-to=type=gha,scope=${{ matrix.target }}
-            *.output=type=local,dest=./dist,platform-split=false
+            *.output=type=local,dest=./dist
       -
-        name: List artifacts
+        name: Rename provenance
         run: |
+          for pdir in ./dist/*/; do
+            (
+              cd "$pdir"
+              binname=$(find . -name '*.tar.gz')
+              filename=$(basename "${binname%.tar.gz}")
+              mv "provenance.json" "${filename}.provenance.json"
```

Comment:
Can't we set the provenance filename with build request instead of renaming later?

Reply:
It works with https://github.com/tonistiigi/xx/pull/141/commits/5433bff9c48af90fbb87c4390d95ee692f2ca48d but filenames are not aligned so needs to renamed them anyway :persevere:: https://github.com/tonistiigi/xx/actions/runs/7783168263/job/21221059915?pr=141#step:7:7

![image](https://github.com/tonistiigi/xx/assets/1951866/35df1296-b8e2-49e3-acf1-7e5cfe1628f0)

I revert this change, I think my initial attempt looks better: https://github.com/tonistiigi/xx/pull/141#issuecomment-1889021879